test/mutable_arithmetics.jl

```diff
@@ -126,7 +126,10 @@ end
 function test_DenseAxisVector(ModelType, ::Any)
     model = ModelType()
     @variable(model, y[2:5])
-    return MA.Test.array_test(y, exclude = ["matrix_vector", "non_array"])
+    return MA.Test.array_test(
+        y,
+        exclude = ["matrix_vector", "non_array", "dot"],
```

Comment:
@blegat, I had to exclude the `dot` test because of this test:
https://github.com/jump-dev/MutableArithmetics.jl/blob/595460dd664380e7668624fccb4f26a637fc1af6/src/Test/array.jl#L284

Which is effectively:
```Julia
model = Model()
@variable(model, x[2:3])
julia> repeat(x, outer = (1, 2))
ERROR: KeyError: key 1 not found
Stacktrace:
  [1] getindex
    @ ./dict.jl:482 [inlined]
  [2] lookup_index
    @ ~/.julia/dev/JuMP/src/Containers/DenseAxisArray.jl:145 [inlined]
  [3] _to_index_tuple
    @ ~/.julia/dev/JuMP/src/Containers/DenseAxisArray.jl:154 [inlined]
  [4] to_index
    @ ~/.julia/dev/JuMP/src/Containers/DenseAxisArray.jl:174 [inlined]
  [5] getindex
    @ ~/.julia/dev/JuMP/src/Containers/DenseAxisArray.jl:192 [inlined]
  [6] getindex
    @ ./reshapedarray.jl:229 [inlined]
  [7] iterate
    @ ./abstractarray.jl:1090 [inlined]
  [8] iterate
    @ ./abstractarray.jl:1088 [inlined]
  [9] _unsafe_setindex!(::IndexLinear, ::Matrix{VariableRef}, ::Base.ReshapedArray{VariableRef, 2, JuMP.Containers.DenseAxisArray{VariableRef, 1, Tuple{UnitRange{Int64}}, Tuple{Dict{Int64, Int64}}}, Tuple{}}, ::UnitRange{Int64}, ::UnitRange{Int64})
    @ Base ./multidimensional.jl:897
 [10] _setindex!
    @ ./multidimensional.jl:887 [inlined]
 [11] setindex!
    @ ./abstractarray.jl:1261 [inlined]
 [12] repeat_outer(a::Base.ReshapedArray{VariableRef, 2, JuMP.Containers.DenseAxisArray{VariableRef, 1, Tuple{UnitRange{Int64}}, Tuple{Dict{Int64, Int64}}}, Tuple{}}, ::Tuple{Int64, Int64})
    @ Base._RepeatInnerOuter ./abstractarraymath.jl:339
 [13] repeat_inner_outer
    @ ./abstractarraymath.jl:327 [inlined]
 [14] repeat(arr::JuMP.Containers.DenseAxisArray{VariableRef, 1, Tuple{UnitRange{Int64}}, Tuple{Dict{Int64, Int64}}}; inner::Nothing, outer::Tuple{Int64, Int64})
    @ Base._RepeatInnerOuter ./abstractarraymath.jl:269
 [15] #repeat#188
    @ ./abstractarraymath.jl:261 [inlined]
 [16] top-level scope
    @ REPL[26]:1
```

Reply:
On Julia v1.5, we have
```julia
julia> x = Containers.@container(x[2:3], 1)
1-dimensional DenseAxisArray{Int64,1,...} with index sets:
    Dimension 1, 2:3
And data, a 2-element Array{Int64,1}:
 1
 1

julia> repeat(x, 1, 2)
2×2 Array{Int64,2}:
 1  1
 1  1
```

Reply:
Arguably, this is incorrect. It should be `repeat(x.data, outer=(1,2))`. Why would repeating a `DenseAxisArray` create a matrix? Does anyone use `repeat` in production code? None of the tests obviously do.

Reply:
cc @mlubin thoughts on this breaking?

Reply:
I'm similarly confused about why `repeat` would return an `Array` given a `DenseAxisArray`. However, changing the behavior is clearly a breaking change. We have no ability to tell who's using which methods.

Reply:
I've added a patch to restore the old behavior.